lib/bundler/resolver.rb

```diff
@@ -124,23 +124,17 @@ def source
 
       def for?(platform, ruby_version)
         spec = @specs[platform]
-        return false unless spec
-
-        return true if ruby_version.nil?
-        # Only allow endpoint specifications since they won't hit the network to
-        # fetch the full gemspec when calling required_ruby_version
-        return true if !spec.is_a?(EndpointSpecification) && !spec.is_a?(Gem::Specification)
-        return true if spec.required_ruby_version.nil?
-
-        spec.required_ruby_version.satisfied_by?(ruby_version.to_gem_version_with_patchlevel)
+        !spec.nil?
```

Comment:
looks like this entire method should probably be removed, and usages of `for?` should be replaced with `if @specs[platform]`?